include/zephyr/sys/device_mmio.h

```diff
@@ -47,7 +47,7 @@
 #include <zephyr/sys/mem_manage.h>
 #include <zephyr/sys/sys_io.h>
 
-#ifdef DEVICE_MMIO_IS_IN_RAM
+#if defined(DEVICE_MMIO_IS_IN_RAM) || defined(__DOXYGEN__)
```

Comment:
Doxygen doc is already built with `CONFIG_MMU` set, so this shouldn't be necessary? `device_map()` is already showing up, for example.
What did you mean by "Add __DOXYGEN__ to include another struct and function."?

Reply:
I was kind of on autopilot when I see a #ifdef, I added __DOXYGEN__ to make sure they appear.

Removed since it is showing up without adding __DOXYGEN__.